src/components/MapView/Legends/index.tsx

```diff
@@ -231,58 +236,59 @@ function LegendItem({
     }
   };
 
-  const boundaryLayer = getBoundaryLayerSingleton();
-  const boundaryLayerData = useSelector(layerDataSelector(boundaryLayer.id)) as
-    | LayerData<BoundaryLayerProps>
-    | undefined;
-
   const legendLayerData = useSelector(layerDataSelector(layer.id)) as
     | LayerData<LayerType>
     | undefined;
 
-  const { startDate: selectedDate } = useSelector(dateRangeSelector);
-
-  const downloadToFile = (
-    source: { content: string; isUrl: boolean },
-    fileName: string,
-    contentType: string,
-  ) => {
-    const link = document.createElement('a');
-    link.setAttribute(
-      'href',
-      source.isUrl
-        ? source.content
-        : URL.createObjectURL(
-            new Blob([source.content], { type: contentType }),
-          ),
-    );
-
-    link.setAttribute('download', fileName);
-    link.click();
-  };
-
   const handleLayerDownload = (
     legendLayer: LayerType,
     e: React.ChangeEvent<{}>,
-  ) => {
+  ): void => {
     e.preventDefault();
-    const extent = bbox(boundaryLayerData?.data) as Extent;
 
     switch (legendLayer.type) {
       case 'wms': {
-        const dataUrl = getWCSLayerUrl({
-          layer: legendLayer,
-          extent,
-          date: selectedDate,
-        });
-        downloadToFile(
-          {
-            content: dataUrl,
-            isUrl: true,
-          },
-          legendLayer.serverLayerName,
-          'image/tiff',
-        );
+        const hasGeometry = legendLayer.geometry !== undefined;
+        if (hasGeometry) {
+          const dataUrl = getWFSLayerUrl({
+            layer: legendLayer,
+            extent,
+            date: selectedDate,
+          });
+          fetch(dataUrl)
+            .then(res => res.json())
+            .then(data => {
+              downloadToFile(
+                {
+                  content: JSON.stringify(data),
+                  isUrl: false,
+                },
+                legendLayer.title,
+                'application/json',
+              );
+            })
+            .catch(err => {
+              throw new Error(err);
+            });
```

Comment:
```suggestion
            .catch(console.error);
```
I'm not too sure if this is needed since the error will be thrown regardless if there's no catch case. If you want you can pass it into `console.error` which will be neater than throwing the error (unhandled errors in promises are discouraged: https://developers.redhat.com/blog/2016/11/09/node-7-and-promise-rejections-please-handle-them)

Reply:
You are right. Thank you for the link was not aware. Fix below as you exactly suggested.

```typescript
...catch(console.error);
```